// [[Reason: Invalid computed property can only be detected by TSC]] ////

//// [tests/cases/compiler/bigintIndex.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -12,9 +12,11 @@
 
 //// [b.d.ts]
 declare const a: {};
 declare const b: invalid;
-declare const c: invalid;
+declare const c: {
+    [bigNum]: number;
+};
 
 /// [Errors] ////
 
 a.ts(2,6): error TS1268: An index signature parameter type must be 'string', 'number', 'symbol', or a template literal type.
@@ -27,9 +29,8 @@
 b.ts(2,19): error TS1128: Declaration or statement expected.
 b.ts(3,12): error TS2464: A computed property name must be of type 'string', 'number', 'symbol', or 'any'.
 b.ts(3,12): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 b.ts(4,12): error TS2464: A computed property name must be of type 'string', 'number', 'symbol', or 'any'.
-b.ts(4,12): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 
 
 ==== a.ts (5 errors) ====
     interface BigIntIndex<E> {
@@ -65,9 +66,9 @@
     typedArray["1"] = 0xBB;
     typedArray[2] = 0xCC;
     
     // {1n: 123} is a syntax error; must go in separate file so BigIntIndex error is shown
-==== b.ts (7 errors) ====
+==== b.ts (6 errors) ====
     // BigInt cannot be used as an object literal property
     const a = {1n: 123};
                ~~
 !!! error TS1136: Property assignment expected.
@@ -82,7 +83,5 @@
 !!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
     const c = {[bigNum]: 789};
                ~~~~~~~~
 !!! error TS2464: A computed property name must be of type 'string', 'number', 'symbol', or 'any'.
-               ~~~~~~~~
-!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
     
\ No newline at end of file
